bambi/models.py

```diff
@@ -746,9 +746,9 @@ def __init__(self, name, data, categorical=False, random=False, prior=None,
 
         # For the sake of computational efficiency (i.e., to avoid lots of
         # large matrix multiplications in the backends), invert the dummy-
-        # coding process and represent binary categoricals as a vector of
+        # coding process and represent full-rank dummies as a vector of
         # indices into the coefficients.
-        if data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
+        if constant and data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
```

Comment:
Hmm... can we keep using the same approach for reduced-rank coding, but just index from 0 below if `not constant`, instead of from 1? Or would that still break something?